### PR TITLE
LPS-188212, LPS-187428 Template Upgrade Process

### PR DESCRIPTION
**Pull request by velosofelipe64**
https://liferay.atlassian.net/browse/LPS-187428
https://liferay.atlassian.net/browse/LPS-188212

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/Template.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.util.regex.Pattern;
+
+/**
+ * @author Felipe Veloso
+ */
+public class Template {
+
+	public Template(
+		String pattern, String patternReplacement, String contextVariable) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/Template.java:25]: @velosofelipe64, please sort the parameters alphabetically

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java [hunk 2]
```diff
@@ -23,28 +24,37 @@
...
+		templates.add(
+			new Template(
+				"\\w+\\s*\\=\\s*.+com\\.liferay\\.portal\\.kernel\\." +
+					"servlet\\.BrowserSnifferUtil\\\"\\)",
+				StringPool.BLANK, "browserSniffer"));
+
+		templates.add(
+			new Template(
+				"\\w+\\s*=\\s*serviceContext\\.getThemeDisplay\\(\\)",
+				StringPool.BLANK, "themeDisplay"));
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java:50]: @velosofelipe64, if the `serviceContext.getThemeDisplay` declaration doesn't generate an issue in 7.4 then we can remove this part for now.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java [hunk 3]
```diff
@@ -23,28 +24,37 @@
...
+
+		for (Template template : templates) {
+			_upgradeDDMTemplates(template);
+			_upgradeFragmentEntries(template);
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java:54]: @velosofelipe64, this way we're still rerunning the SQL queries, therefore in order to run them a single time, we should use this `for` inside the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries`. You may try to make the variable `templates` an attribute of this class, populate it with the templates information and get it inside the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries`.

Does it make sense to you?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java (hunk 3 above)
Comment (velosofelipe64) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateUpgradeProcess.java:54]: Yes, sure.
Do you think we can simplify the methods `_upgradeDDMTemplates` and `_upgradeFragmentEntries` in only one, with the variable `templates` and `selectPreparedStatement` as parameters?

**Pull request closed without merging**